src/main/java/bookstore/controller/OrderController.java

```diff
@@ -0,0 +1,116 @@
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/orders")
+@RequiredArgsConstructor
+@ControllerAdvice
+@Tag(name = "Order Controller", description = "Endpoints for managing orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final OrderItemService orderItemService;
+    private final SortParametersParsingUtil sortParametersParsingUtil;
+    
+    @Operation(summary = "Place a new order")
+    @ResponseStatus(HttpStatus.CREATED)
+    @ApiResponse(responseCode = "201", description = "Order created", content = {
+            @Content(mediaType = "application/json", schema =
+            @Schema(implementation = OrderResponseDto.class),
+            examples = @ExampleObject(value = "{\n"
+                                              + "    \"id\": 1,\n"
+                                              + "    \"shippingAddress\": \"address\",\n"
+                                              + "    \"orderItems\": [\n"
+                                              + "      {\n"
+                                              + "        \"bookId\": 1,\n"
+                                              + "        \"quantity\": 2\n"
+                                              + "      }\n"
+                                              + "    ]\n"
+                                              + "}"))
+    })
+    @PostMapping
+    public OrderResponseDto placeOrder(@Valid @RequestBody OrderRequestDto order,
+                                       Authentication authentication) {
+        return orderService.save(order, (User) authentication.getPrincipal());
+    }
+
+    @Operation(summary = "Get all orders")
+    @GetMapping
+    public List<OrderResponseDto> getAllOrders(Authentication authentication,
+                                               @Parameter(description =
+                                                   "Page number (zero-based)", example = "0")
+            @RequestParam(defaultValue = "0") int page,
+                                               @Parameter(description =
+                                                   "Number of items per page", example = "10")
+                                                   @RequestParam(defaultValue = "10") int size,
+                                               @Parameter(description =
+                                                   "Sort order and fields",
+                                                   example = "orderDate,desc")
+                                                   @RequestParam(defaultValue = "orderDate,desc")
+                                                       String sort) {
+        Sort.Order sortOrder = sortParametersParsingUtil.parseSortOrder(sort);
+
+        Pageable pageable = PageRequest.of(page, size, Sort.by(sortOrder));
+        return orderService.getByUser((User) authentication.getPrincipal(), pageable);
```

Comment:
Controller should only invoke the Service' methods.

Logic of parsing order, making pageable should be inside service layer.